specs/simple-serialize.md

```diff
@@ -33,6 +33,8 @@ This is a **work in progress** describing typing, serialization and Merkleizatio
 
 ### Composite types
 
+Composite types are limited to a maximum of `2**32 - 1` elements.
```

Comment:
For Merkleisation purposes this is unnecessarily strict. We want this limit for lists (because of the length mixing), but *not* for containers and vectors. One example use case is that we want to represent the sparse Merkle trees in shards as vectors of length 2^256.

Reply:
I feel like I had a reason for this I'm failing to remember it.  Removing unless I remember in the next few minutes.